dao/FashionKnex.js

```diff
@@ -27,35 +27,66 @@ class FashionKnex {
     this._knex = knexConnection || knex(knexConfig)
   }
 
-  async insert (data, returning) {
+  /**
+   * TODO: replace with https://github.com/tgriesser/knex/pull/2197 after new
+   * and improved implementation is submitted
+   * Ref: https://gist.github.com/plurch/118721c2216f77640232
+   * http://www.postgresql.org/docs/9.5/static/sql-insert.html
+   */
+  async upsert (data, {conflictColumn = 'id', returning = '*'} = {}) {
     data = _clean(data)
-    returning = returning || '*'
 
-    let result
+    const insertString = this._knex
+      .insert(data)
+      .into(this._tableName)
+      .toString()
+    const conflictString = this._knex.raw(` ON CONFLICT (??) DO NOTHING`,
+      conflictColumn).toString()
+    const query = (insertString + conflictString + ` RETURNING ${returning}`)
+      .replace(/\?/g, '\\?')
+
+    return this._knex.raw(query)
```

Comment:
Why not use `async/await` with a `try/catch`?

Reply:
I assume it's because you copied this code from the other repo.

Reply:
I got rid of the try/catch here because I don't need to do anything with the result of the query. If there were post-operation work, it'd make more sense to wrap in a try/catch.

Reply:
update: I take back what I said - I'd say this is a good candidate for a try/catch since I'm doing something with the result here. Will update.